compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -420,8 +444,14 @@ pub fn llvm_global_features(sess: &Session) -> Vec<String> {
     features.extend(sess.target.features.split(',').flat_map(&filter));
 
     // -Ctarget-features
-    features.extend(sess.opts.cg.target_feature.split(',').flat_map(&filter));
-
+    let feats: Vec<&str> = sess.opts.cg.target_feature.split(',').collect();
+    // LLVM enables based on the last occurence of a feature
+    if let Some(f) =
+        check_tied_features(sess, &feats.iter().map(|f| (strip(f), !f.starts_with("-"))).collect())
+    {
+        sess.err(&format!("Target features {} must all be enabled or disabled", f.join(", ")));
```

Comment:
```suggestion
        sess.err(&format!("Target features {} must all be enabled or disabled together", f.join(", ")));
```
Slight tweak to the message that makes it clearer.